src/test/java/TopNavigationTest.java

```diff
@@ -0,0 +1,150 @@
+    private static final By topNavigationBarCategoryNameSecondLevelForWomen = By.cssSelector(".navigation li.level0:nth-child(2) ul.level0 li.level1:nth-child(1) .level2 a > span:not(.ui-menu-icon)");
+    private static final By topNavigationBarCategorySecondLevelForWomenLink = By.cssSelector(".navigation li.level0:nth-child(2) ul.level0 li.level1:nth-child(1) .level2 a");
+    private final By topNavigationBarSecondLevelForWomanTops = By.cssSelector(".navigation li.level0:nth-child(2) ul.level0 li.level1:nth-child(1)");
+
+    private WebDriver driver;
+    private TopNavigation topNavigation;
+    private List<WebElement> topCategoriesList;
+    private List<WebElement> secondLevelTopCategoriesList;
+
+    @BeforeClass
+    public void setUp() {
+        driver = WebDriverUtils.createWebDriver();
+        driver.get(HOMEPAGE_URL);
+        topNavigation = new TopNavigation(driver);
+        List<WebElement> topCategoriesList;
+        List<WebElement> secondLevelTopCategoriesList;
+    }
+
+    @Test
+    public void topNavigation_TopLevelCategories_CorrectlyDisplayed() {
+        //given
+        topCategoriesList = topNavigation.getTopCategoriesList();
+
+        //then
+        List<String> expectedNamesList = List.of("What's New", "Women", "Men", "Gear", "Training", "Sale");
+        List<String> expectedLinkList = List.of("what-is-new.html", "women.html", "men.html",
+                "gear.html", "training.html", "sale.html");
+        validateMenuCategoriesNamesAndLinks(topCategoriesList, expectedNamesList, expectedLinkList);
+    }
+
+    @DataProvider
+    public Object[][] secondCategoriesLevelWithExpectedNamesAndLinks() {
+        topCategoriesList = topNavigation.getTopCategoriesList();
+
+        return new Object[][]{
+                {topCategoriesList.get(1), List.of("Tops", "Bottoms"), List.of("women/tops-women.html", "women/bottoms-women.html")},
+                {topCategoriesList.get(2), List.of("Tops", "Bottoms"), List.of("men/tops-men.html", "men/bottoms-men.html")},
+                {topCategoriesList.get(3), List.of("Bags", "Fitness Equipment", "Watches"), List.of("gear/bags.html", "gear/fitness-equipment.html",
+                        "gear/watches.html")},
+                {topCategoriesList.get(4), List.of("Video Download"), List.of("training/training-video.html")},
+        };
+    }
+
+    @Test(dataProvider = "secondCategoriesLevelWithExpectedNamesAndLinks")
+    public void topNavigation_SecondLevelCategories_CorrectlyDisplayed(WebElement topCategoryHavingSecondLevelSubMenu, List<String> expectedCategoriesNames,
+                                                                       List<String> expectedCategoriesLinks) {
+        //when
+        topNavigation.hoverOnCategory(topCategoryHavingSecondLevelSubMenu);
+        List<WebElement> secondLevelCategories = topCategoryHavingSecondLevelSubMenu.findElements(By.cssSelector("ul.level0 li.level1:nth-child(1)"));
```

Comment:
Invalid selector. Tests works only for first category having submenu.
Selector should be "ul.level0 li.level1"